src/main/java/org/openelisglobal/notification/dao/NotificationPayloadTemplateDAOImpl.java

```diff
@@ -22,10 +23,20 @@ public NotificationPayloadTemplateDAOImpl() {
     public NotificationPayloadTemplate getSystemDefaultPayloadTemplateForType(NotificationPayloadType type) {
         List<NotificationPayloadTemplate> data;
         try {
-            String sql = "from NotificationPayloadTemplate as npt where npt.type = :type order by npt.id asc";
-            Query<NotificationPayloadTemplate> query = entityManager.unwrap(Session.class).createQuery(sql,
+            String jpql = "SELECT npt FROM NotificationPayloadTemplate npt WHERE npt.type = :type ORDER BY npt.id"
+                    + " ASC";
+            Query<NotificationPayloadTemplate> query = entityManager.unwrap(Session.class).createQuery(jpql,
                     NotificationPayloadTemplate.class);
-            query.setParameter("type", type.name());
+
+            // Ensure type is set correctly according to PostgreSQL's expectations
+            if (entityManager.getEntityManagerFactory().getProperties().get("hibernate.dialect").toString()
```

Comment:
Different databases have varying requirements for handling enums; PostgreSQL, for example, requires `enums` to be passed as strings, unlike some other databases. Using `StringType.INSTANCE` ensures Hibernate treats the parameter correctly, reducing the risk of type-related errors in this query.